streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -623,16 +613,46 @@ void runOnce() {
             return;
         }
 
-        initializeAndRestorePhase();
+        // we need to first add closed tasks and then created tasks to work with those revived / recycled tasks
+        restoreThread.addClosedTasks(taskManager.drainRemovedTasks());
+
+        // try to initialize created tasks that are either newly assigned or re-created from corrupted tasks
+        final List<AbstractTask> initializedTasks;
+        if (!(initializedTasks = taskManager.tryInitializeNewTasks()).isEmpty()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Initializing newly created tasks {} under state {}",
+                        initializedTasks.stream().map(AbstractTask::id).collect(Collectors.toList()), state);
+            }
+
```

Comment:
I think it's worth logging an INFO-level message here just saying `All tasks initialized and started restoring` or something

Reply:
Ack.